OnBaseDocsApi/Controllers/OnBaseDocsController.cs

```diff
@@ -57,7 +60,188 @@ public IHttpActionResult GetFile(int id)
+                            return BadRequestResult("More than one document attributes was included.");
+
+                        docAttr = JObject
+                            .Parse(content.ReadAsStringAsync().Result)
+                            .ToObject<DocumentPostAttributes>();
+                    }
+                    else
+                    {
+                        return BadRequestResult($"Unexpected Content-Disposition header of name '{dispoName}'.");
+                    }
+                }
+            }
+            catch (JsonReaderException ex)
+            {
+                return BadRequestResult($"JSON parse error. {ex.Message}");
+            }
+            catch (IOException ex) when (ex.Message.Contains("MIME multipart message is not complete"))
+            {
+                return BadRequestResult(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                return InternalErrorResult(ex.Message);
+            }
+
+            if ((docStream == null) || string.IsNullOrWhiteSpace(docExt))
+                return BadRequestResult("The required parameter 'file' is missing.");
+            if (docAttr == null)
+                return BadRequestResult("The required parameter 'attributes' is missing.");
+            if (string.IsNullOrWhiteSpace(docAttr.IndexKey))
+                return BadRequestResult("The required parameter 'IndexKey' is empty.");
+            if (string.IsNullOrWhiteSpace(docAttr.DocumentType))
+                return BadRequestResult("The required parameter 'DocumentType' is empty.");
+
+            return TryHandleRequest(app =>
+            {
+                // The document is moved to staging when there are no keywords included
+                // so that we can kick off a re-index to generate the autofill keywords.
+                bool toStaging = (docAttr.Keywords == null) || !docAttr.Keywords.Any();
+
+                // When going to staging, confirm that the final doc type is actually valid.
+                DocumentType finalDocType = null;
+                if (toStaging)
+                {
+                    finalDocType = app.Core.DocumentTypes.Find(docAttr.DocumentType);
+                    if (finalDocType == null)
+                        return BadRequestResult($"The DocumentType '{docAttr.DocumentType}' could not be found.");
+                }
+
+                var createAttr = new DocumentCreateAttributes
+                {
+                    DocumentType = toStaging ? Global.Config.StagingDocType : docAttr.DocumentType,
+                    Comment = docAttr.Comment,
+                    IndexKey = docAttr.IndexKey,
+                    Keywords = docAttr.Keywords,
+                    Ext = docExt,
+                    Stream = docStream,
+                    ToStaging = toStaging,
+                };
+                var result = CreateDocument(app, createAttr, out var doc);
+                if (result != null)
```

Comment:
Correct me if I'm wrong, looks like `CreateDocument` will return `null` only if everything looks good? If so, I think `result` here is misleading; we should probalbly name it `errors` or something else.

Reply:
Good point.